.agents/conventions.md

```
@@ -70,12 +70,16 @@ export const ParamKind = {
     Query: 'query',
     // ...
 } as const;
-export type ParamKindValue = `${typeof ParamKind[keyof typeof ParamKind]}`;
+// Same-name type alias: lets `${ParamKind}` template-literal types
+// resolve to the value union (matching the prior enum ergonomic).
+export type ParamKind = typeof ParamKind[keyof typeof ParamKind];
 ```
 
-- **Public API consumers can use either form.** `ParamKind.Body` is more discoverable and survives renames; the literal `'body'` is more concise. The type accepts both.
+- **Use the same name for the const and the type alias.** Value namespace and type namespace are separate in TypeScript — `ParamKind` (value) is the const object, `ParamKind` (type) is the value union. Don't introduce a `*Value` suffix.
+- **Reference the type via `\`${ParamKind}\``** when you want the bare-string union. Reference `ParamKind.Body` when you want the named constant.
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)